chain/client/src/sync/state.rs

```diff
@@ -453,6 +464,15 @@ impl StateSync {
         self.resharding_state_roots.insert(shard_id, result);
     }
 
+    // Called by the client actor, when it finished loading memtrie.
+    pub fn set_load_memtrie_result(
+        &mut self,
+        shard_id: ShardId,
+        result: Result<(), near_chain::Error>,
+    ) {
+        self.load_memtrie_results.insert(shard_id, result);
```

Comment:
Not sure if it makes sense to check if there were any existing values here? I know we are not doing it for the other "result" fields but can consider.

I see that we are removing the value later in the next step

Reply:
We could but I am not sure what to do in such case anyway. We could panic or print an error.

Reply:
I think we should be panicking here as this would be a code issue, but seems like the rest of the codebase doesn't follow convention we can skip it.